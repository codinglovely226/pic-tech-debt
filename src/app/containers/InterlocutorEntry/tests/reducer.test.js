/* eslint-disable */

import {
  addCallRecord,
  addCase,
  updateCaseStatus,
} from 'containers/CaseEntry/actions/case';

import InterlocutorEntryReducer from '../reducer';
import {
  replaceCommunications,
  addCommunication,
} from '../actions';

describe('InterlocutorEntryReducer', () => {
  var mockedDate = new Date(2018, 2, 4, 4, 59, 13, 805);
  global.Date = jest.fn(() => mockedDate);
  it('returns the initial state', () => {
    expect(InterlocutorEntryReducer(undefined, {})).toMatchSnapshot();
  });
  it('adds a communication record', () => {
    expect(InterlocutorEntryReducer({}, addCommunication(1, { user: 'test' }))).toMatchSnapshot();
  });
  it('adds a call record', () => {
    const action = addCallRecord(1, { user: 'test' });
    action.commsNoteId = 'de6d38464f8ee145f0731aca7faa24f78d07e25f';
    expect(InterlocutorEntryReducer({}, action)).toMatchSnapshot();
  });
  it('adds a case', () => {
    const action = addCase({ user: 'test' });
    action.commsNoteId = 'de6d38464f8ee145f0731aca7faa24f78d07e25f';
    expect(InterlocutorEntryReducer({}, action)).toMatchSnapshot();
  });
  it('replaces communication records', () => {
    expect(InterlocutorEntryReducer({}, replaceCommunications(1, [{}, {}]))).toMatchSnapshot();
  });
  it('update case status', () => {
    expect(InterlocutorEntryReducer({}, updateCaseStatus(1, 'open', { user: 'test' }, false))).toMatchSnapshot();
  });
});
