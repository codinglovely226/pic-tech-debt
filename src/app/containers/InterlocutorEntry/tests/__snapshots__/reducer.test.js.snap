exports[`InterlocutorEntryReducer adds a call record 1`] = `
Array [
  Object {
    "deleted": false,
    "direction": "incoming",
    "endTimestamp": null,
    "history": Object {
      "agents": Object {},
      "features": Object {
        "descriptors": "",
        "outcome": "",
        "pastDescriptors": Object {},
        "when": "",
      },
      "investigations": Object {},
      "patient": Object {
        "age": null,
        "ageGroup": null,
        "ageUnits": null,
        "ageYears": null,
        "animalType": null,
        "category": null,
        "circumstance": null,
        "gender": null,
        "localityAddress1": null,
        "localityCity": null,
        "localityCountry": null,
        "localityGeometry": null,
        "localityISOCountry": null,
        "localityLocation": null,
        "localityName": null,
        "localityPlaceId": null,
        "localityPostcode": null,
        "localityState": null,
        "localityTelephone": null,
        "medicalRecordNumber": null,
        "medicalRecordNumberEnc": null,
        "name": null,
        "nameEnc": null,
        "occupation": null,
        "outcome": null,
        "poisonSeverityScore": null,
        "pregnantLactating": null,
        "pregnantStatus": null,
        "pregnantTrimester": null,
        "pregnantTrimesterUnknown": null,
        "riskAssessment": null,
        "topic": null,
        "weight": null,
        "weightUnits": null,
      },
      "treatments": Object {
        "pastTreatments": Object {},
        "referrals": Array [],
        "treatments": Array [],
      },
    },
    "interlocutor": Object {
      "category": "",
      "contact": "",
      "name": "",
      "nameEnc": "",
      "organisation": Object {
        "address1": "",
        "address2": "",
        "city": "",
        "country": "",
        "geometry": Object {},
        "isoCountry": "",
        "location": "",
        "name": "",
        "placeId": "",
        "postcode": "",
        "state": "",
        "telephone": "",
      },
    },
    "medium": Object {
      "detail": "",
      "detailEnc": "",
      "type": "Phone",
    },
    "noteRef": "de6d38464f8ee145f0731aca7faa24f78d07e25f",
    "sequenceId": 1,
    "spi": Object {
      "id": undefined,
      "job": undefined,
      "name": undefined,
      "org": undefined,
      "roles": undefined,
      "siteId": undefined,
    },
    "timestamp": "2018-03-04T04:59:13.805Z",
    "type": "comms",
  },
]
`;

exports[`InterlocutorEntryReducer adds a case 1`] = `
Array [
  Object {
    "deleted": false,
    "direction": "incoming",
    "endTimestamp": null,
    "history": Object {
      "agents": Object {},
      "features": Object {
        "descriptors": "",
        "outcome": "",
        "pastDescriptors": Object {},
        "when": "",
      },
      "investigations": Object {},
      "patient": Object {
        "age": null,
        "ageGroup": Array [
          null,
          "adult",
        ],
        "ageUnits": null,
        "ageYears": null,
        "animalType": null,
        "category": Array [
          null,
          "human",
        ],
        "circumstance": null,
        "gender": null,
        "localityAddress1": null,
        "localityCity": null,
        "localityCountry": null,
        "localityGeometry": Array [
          null,
          Object {},
        ],
        "localityISOCountry": null,
        "localityLocation": null,
        "localityName": null,
        "localityPlaceId": null,
        "localityPostcode": null,
        "localityState": null,
        "localityTelephone": null,
        "medicalRecordNumber": null,
        "medicalRecordNumberEnc": null,
        "name": null,
        "nameEnc": null,
        "occupation": Array [
          null,
          "unknown",
        ],
        "outcome": null,
        "poisonSeverityScore": Array [
          null,
          "Moderate",
        ],
        "pregnantLactating": Array [
          null,
          "unknown",
        ],
        "pregnantStatus": Array [
          null,
          "unknown",
        ],
        "pregnantTrimester": Array [
          null,
          "unknown",
        ],
        "pregnantTrimesterUnknown": null,
        "riskAssessment": Array [
          null,
          "Moderate Risk",
        ],
        "topic": null,
        "weight": null,
        "weightUnits": Array [
          null,
          "kg",
        ],
      },
      "treatments": Object {
        "pastTreatments": Object {},
        "referrals": Array [],
        "treatments": Array [],
      },
    },
    "interlocutor": Object {
      "category": "",
      "contact": "",
      "name": "",
      "nameEnc": "",
      "organisation": Object {
        "address1": "",
        "address2": "",
        "city": "",
        "country": "",
        "geometry": Object {},
        "isoCountry": "",
        "location": "",
        "name": "",
        "placeId": "",
        "postcode": "",
        "state": "",
        "telephone": "",
      },
    },
    "medium": Object {
      "detail": "",
      "detailEnc": "",
      "type": "Phone",
    },
    "noteRef": "de6d38464f8ee145f0731aca7faa24f78d07e25f",
    "sequenceId": 1,
    "spi": Object {
      "id": undefined,
      "job": undefined,
      "name": undefined,
      "org": undefined,
      "roles": undefined,
      "siteId": undefined,
    },
    "timestamp": "2018-03-04T04:59:13.805Z",
    "type": "comms",
  },
]
`;

exports[`InterlocutorEntryReducer adds a communication record 1`] = `
Array [
  Object {
    "deleted": false,
    "direction": "incoming",
    "endTimestamp": null,
    "history": Object {
      "agents": Object {},
      "features": Object {
        "descriptors": "",
        "outcome": "",
        "pastDescriptors": Object {},
        "when": "",
      },
      "investigations": Object {},
      "patient": Object {
        "age": null,
        "ageGroup": null,
        "ageUnits": null,
        "ageYears": null,
        "animalType": null,
        "category": null,
        "circumstance": null,
        "gender": null,
        "localityAddress1": null,
        "localityCity": null,
        "localityCountry": null,
        "localityGeometry": null,
        "localityISOCountry": null,
        "localityLocation": null,
        "localityName": null,
        "localityPlaceId": null,
        "localityPostcode": null,
        "localityState": null,
        "localityTelephone": null,
        "medicalRecordNumber": null,
        "medicalRecordNumberEnc": null,
        "name": null,
        "nameEnc": null,
        "occupation": null,
        "outcome": null,
        "poisonSeverityScore": null,
        "pregnantLactating": null,
        "pregnantStatus": null,
        "pregnantTrimester": null,
        "pregnantTrimesterUnknown": null,
        "riskAssessment": null,
        "topic": null,
        "weight": null,
        "weightUnits": null,
      },
      "treatments": Object {
        "pastTreatments": Object {},
        "referrals": Array [],
        "treatments": Array [],
      },
    },
    "interlocutor": Object {
      "category": "",
      "contact": "",
      "name": "",
      "nameEnc": "",
      "organisation": Object {
        "address1": "",
        "address2": "",
        "city": "",
        "country": "",
        "geometry": Object {},
        "isoCountry": "",
        "location": "",
        "name": "",
        "placeId": "",
        "postcode": "",
        "state": "",
        "telephone": "",
      },
    },
    "medium": Object {
      "detail": "",
      "detailEnc": "",
      "type": "Phone",
    },
    "noteRef": null,
    "sequenceId": 1,
    "spi": Object {
      "id": undefined,
      "job": undefined,
      "name": undefined,
      "org": undefined,
      "roles": undefined,
      "siteId": undefined,
    },
    "timestamp": "2018-03-04T04:59:13.805Z",
    "type": "comms",
  },
]
`;

exports[`InterlocutorEntryReducer replaces communication records 1`] = `
Array [
  Object {},
  Object {},
]
`;

exports[`InterlocutorEntryReducer returns the initial state 1`] = `Array []`;

exports[`InterlocutorEntryReducer update case status 1`] = `
Array [
  Object {
    "history": Object {
      "agents": Object {},
      "features": Object {
        "descriptors": "",
        "outcome": "",
        "pastDescriptors": Object {},
        "when": "",
      },
      "investigations": Object {},
      "patient": Object {
        "age": null,
        "ageGroup": null,
        "ageUnits": null,
        "ageYears": null,
        "animalType": null,
        "category": null,
        "circumstance": null,
        "gender": null,
        "localityAddress1": null,
        "localityCity": null,
        "localityCountry": null,
        "localityGeometry": null,
        "localityISOCountry": null,
        "localityLocation": null,
        "localityName": null,
        "localityPlaceId": null,
        "localityPostcode": null,
        "localityState": null,
        "localityTelephone": null,
        "medicalRecordNumber": null,
        "medicalRecordNumberEnc": null,
        "name": null,
        "nameEnc": null,
        "occupation": null,
        "outcome": null,
        "poisonSeverityScore": null,
        "pregnantLactating": null,
        "pregnantStatus": null,
        "pregnantTrimester": null,
        "pregnantTrimesterUnknown": null,
        "riskAssessment": null,
        "topic": null,
        "weight": null,
        "weightUnits": null,
      },
      "treatments": Object {
        "pastTreatments": Object {},
        "referrals": Array [],
        "treatments": Array [],
      },
    },
    "newState": "open",
    "spi": Object {
      "id": undefined,
      "job": undefined,
      "name": undefined,
      "org": undefined,
      "roles": undefined,
      "siteId": undefined,
    },
    "timestamp": "2018-03-04T04:59:13.805Z",
    "type": "status",
  },
]
`;
